spec/controllers/hq/partners_controller_spec.rb

```diff
@@ -16,4 +16,29 @@ def foobar_action
       get :foobar_action
     end
   end
+
+  describe 'while signed-in' do
+    before :each do
+      sign_in instance_double(AdminUser)
+    end
+
+    describe 'GET #show' do
+      before :each do
+        FactoryGirl.create(:partner)
+      end
+
+      example 'invalid :id should redirect & set flash' do
+        get :show, id: (Partner.all.count + 1)
```

Comment:
agreed in addition id's in database aren't guaranteed to be less than the record count. There are better ways to setup Partner not being found.